projects/material-addons/src/lib/layout/sidebar-layout/sidebar-layout.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+import { CommonModule, Location } from '@angular/common';
+import { PortalModule, TemplatePortal } from '@angular/cdk/portal';
+import { ContentPanelModule } from '../content-panel/content-panel.module';
+import {MatIconModule} from '@angular/material/icon';
+import {Router, RouterModule} from '@angular/router';
+
+@Component({
+  selector: 'mad-sidebar-layout',
+  standalone: true,
+  templateUrl: './sidebar-layout.component.html',
+  styleUrl: './sidebar-layout.component.scss',
+  imports: [CommonModule, ContentPanelModule, PortalModule, MatIconModule, RouterModule],
+})
+export class SidebarLayoutComponent {
```

Comment:
Maybe we should extend the @Input decorators with the required property, wherever necessary?

Reply:
Fixed. Added required flag to sideBarItems and content. Because everything else is optional